### PR TITLE
Batch Delete by Filter

### PR DESCRIPTION
**Pull request by antas-marcin**

### CODE REVIEW

File: adapters/repos/db/index.go [hunk 1]
```diff
@@ -993,3 +993,96 @@ func (i *Index) notifyReady() {
...
+				objs = i.remote.DeleteObjectBatch(ctx, shardName, docIDs, dryRun)
+			} else {
+				shard := i.Shards[shardName]
+				objs = shard.deleteObjectBatch(ctx, docIDs, dryRun)
+			}
+			ch <- result{objs}
+		}(shardName, docIDs)
+	}
+
+	wg.Wait()
```
Comment (parkerduckworth) [inline adapters/repos/db/index.go:1067]: I really like this approach of just deleting from each shard in parallel, returning the status/errors for each deleted item.

File: adapters/repos/db/shard_read.go [hunk 2]
```diff
@@ -364,3 +364,71 @@ func (s *Shard) buildAllowList(ctx context.Context, filters *filters.LocalFilter
 
 	return list, nil
 }
+
+func (s *Shard) docIDToUUID(docID uint64) (strfmt.UUID, error) {
```
Comment (parkerduckworth) [inline adapters/repos/db/shard_read.go:368]: I think that `uuidFromDocID` reads a bit better, but that is just my preference, this makes sense too

File: adapters/repos/db/shard_write_batch_delete.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+func (b *deleteObjectsBatcher) delete(ctx context.Context,
+	docIDs []uint64, dryRun bool) {
+	b.objects = b.deleteSingleBatchInLSM(ctx, docIDs, dryRun)
+}
+
+func (b *deleteObjectsBatcher) deleteSingleBatchInLSM(ctx context.Context,
+	batch []uint64, dryRun bool) objects.BatchSimpleObjects {
+	result := make(objects.BatchSimpleObjects, len(batch))
+	objLock := &sync.Mutex{}
```
Comment (parkerduckworth) [inline adapters/repos/db/shard_write_batch_delete.go:62]: does the Mutex need to be addressable?

File: adapters/repos/db/inverted/objects.go [hunk 4]
```diff
@@ -95,7 +95,7 @@ func (a *Analyzer) analyzeIDProp(id strfmt.UUID) (*Property, error) {
 		return nil, errors.Wrap(err, "marshal id prop")
 	}
 	return &Property{
-		Name:         traverser.InternalPropID,
+		Name:         filters.InternalPropID,
```
[the remark this answers is not in the document]
Comment (antas-marcin) [reply adapters/repos/db/inverted/objects.go:98]: yeah, all in one place :) the same code for validation and no need for re-introducing the special properties names

File: adapters/repos/db/shard_write_batch_delete.go (hunk 3 above)
Comment (antas-marcin) [reply adapters/repos/db/shard_write_batch_delete.go:62]: that one I must say that it's copied from the batch create objects method :) so I would leave it :)

File: adapters/repos/db/shard_read.go [hunk 5]
```diff
@@ -364,3 +364,71 @@ func (s *Shard) buildAllowList(ctx context.Context, filters *filters.LocalFilter
...
+		return errors.Wrap(err, "delete object from bucket")
+	}
+
+	err = s.cleanupInvertedIndexOnDelete(existing, docID)
+	if err != nil {
+		return errors.Wrap(err, "delete object from bucket")
+	}
+
+	// in-mem
+	// TODO: do we still need this?
```
[the remark this answers is not in the document]
Comment (parkerduckworth) [reply adapters/repos/db/shard_read.go:426]: ah ok

**Review by parkerduckworth — APPROVED** [review]

File: adapters/repos/db/shard_read.go (hunk 2 above)
Comment (antas-marcin) [reply adapters/repos/db/shard_read.go]: yes, I agree too. done!

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**